### ISSUE TITLE
Differential mode uploading all files from host

### ISSUE THREAD

**Issue by baileygm** [thelastpickle/cassandra-medusa#25]
I've successfully been able to backup to S3 and restore my node data using medusa 0.3.0 but am finding that regardless whether I use mode=full or mode=differential all files are being uploaded from the host
In a simple test case where I simply backup the system keyspaces and run a full backup  first and then a differential backup immediately afterwards I get the following output

[2019-12-02 17:08:31] INFO: Backup done
[2019-12-02 17:08:31] INFO: - Started: 2019-12-02 17:07:28
                        - Started extracting data: 2019-12-02 17:07:32
                        - Finished: 2019-12-02 17:08:31
[2019-12-02 17:08:31] INFO: - Real duration: 0:00:58.868730 (excludes time waiting for other nodes)
[2019-12-02 17:08:31] INFO: - 376 files, 283.62 KB
[2019-12-02 17:08:31] INFO: - 376 files copied from host
[2019-12-02 17:08:31] INFO: - 0 copied from previous backup (test2)

**Comment by adejanovski** [comment]
Hi @baileygm,

indeed, differential backups require to be consecutive so that pre-existing files won't get uploaded. Here you ran a full backup and then a differential, which is valid but doesn't benefit yet from differential improvements.
I suggest you run another differential now and you should see that most files will be taken from your previous differential backup.

**Comment by baileygm** [comment]
I got the same issue when running a subsequent differential backup unfortunately

[2019-12-02 17:21:35] INFO: - 376 files, 284.79 KB
[2019-12-02 17:21:35] INFO: - 376 files copied from host
[2019-12-02 17:21:35] INFO: - 0 copied from previous backup (test3)

**Comment by adejanovski** [comment]
Do you have a custom (non system) keyspace in there with sstables that didn't get compacted away between the backups?

**Comment by adejanovski** [comment]
Hi @baileygm,

I confirm I could reproduce the problem with S3.
I'll work on a fix ASAP.

**Comment by baileygm** [comment]
Excellent thanks,  sorry I missed your previous message

**Comment by adejanovski** [comment]
I just pushed PR #27 with the fix  : 

```
[2019-12-03 09:06:16] INFO: Backup done
[2019-12-03 09:06:16] INFO: - Started: 2019-12-03 09:05:47
                        - Started extracting data: 2019-12-03 09:05:50
                        - Finished: 2019-12-03 09:06:16
[2019-12-03 09:06:16] INFO: - Real duration: 0:00:25.640207 (excludes time waiting for other nodes)
[2019-12-03 09:06:16] INFO: - 353 files, 3.11 GB
[2019-12-03 09:06:16] INFO: - 48 files copied from host
[2019-12-03 09:06:16] INFO: - 305 copied from previous backup (test7)
[2019-12-03 09:06:16] DEBUG: Emitting metrics
[2019-12-03 09:06:16] DEBUG: Done emitting metrics
```

I'll cut out a release once this is merged.

**Comment by baileygm** [comment]
Would it be possible to include my suggested change to "clean_path" in this release (if you agree with the change) as this is causing the process to fail ?
https://github.com/thelastpickle/cassandra-medusa/issues/26

The issue for me is that saved_caches and commitlog are mounted volumes in a docker container so clean_path throws an exception when it tries to delete them
If clean path deleted the contents rather than the folder there would be no problem

**Comment by baileygm** [comment]
I saw that you released version 0.3.1 and can confirm that its working fine now thanks

**Comment by adejanovski** [comment]
Thanks for the quick feedback @baileygm 👍
